src/uagents/context.py

```diff
@@ -97,22 +99,61 @@ def get_message_protocol(self, message_schema_digest) -> Optional[str]:
                     return protocol_digest
         return None
 
+    def get_agents_by_protocol(self, protocol_digest: str) -> list:
+        if not isinstance(protocol_digest, str) or not protocol_digest.startswith(
+            "proto:"
+        ):
+            self.logger.error(f"Invalid protocol digest: {protocol_digest}")
+            return []
+        response = requests.post(
+            url=ALMANAC_API_URL + "search",
+            json={"text": protocol_digest[6:]},
+            timeout=DEFAULT_ENVELOPE_TIMEOUT_SECONDS,
+        )
+        if response.status_code == 200:
+            data = response.json()
+            return [agent["address"] for agent in data if agent["status"] == "local"]
+        return []
+
     async def send(
         self,
         destination: str,
         message: Model,
         timeout: Optional[int] = DEFAULT_ENVELOPE_TIMEOUT_SECONDS,
     ):
         schema_digest = Model.build_schema_digest(message)
-        await self.send_raw(
+        await self._send_raw(
```

Comment:
We probably don't want to make this private because some people are using `ctx.send_raw()` in message handlers.